BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnosticsModule.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+
+    /// <summary>
+    /// Self diagnostics class captures the EventSource events sent by Application Insights
+    /// modules and writes them to local file for internal troubleshooting.
+    /// </summary>
+    public class SelfDiagnosticsModule : IDisposable
```

Comment:
we dont need this to be public, right? The only way user is going to interact with the module is via the config.json file.

Reply:
You are right. I've made it internal and renamed it.